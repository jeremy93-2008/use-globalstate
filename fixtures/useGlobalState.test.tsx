import React from 'react'
import { test, expect, beforeAll } from 'vitest'
import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { ParentComponent } from './components/ParentComponent'

beforeAll(() => {
    render(<ParentComponent />)    
})

test('useGlobalState -> initial state', () => {
    expect(screen.getByTestId('parent-count').innerText).toEqual('Count: 0')
})

test('useGlobalState -> initial state 2', () => {
    expect(screen.getByTestId('parent-count2').innerText).toEqual('Count: 0')
})

test('useGlobalState -> child component', () => {
    expect(screen.getByTestId('child-count').innerText).toEqual('Count: 0')
})

test('useGlobalState -> child component 2', () => {
    expect(screen.getByTestId('child-count2').innerText).toEqual('Count: 0')
})

test('useGlobalState -> increment count', async () => {
    const user = userEvent.setup()
    await user.click(screen.getByTestId('parent-button-increment'))
    expect(screen.getByTestId('parent-count').innerText).toEqual('Count: 1')
})

test('useGlobalState -> increment count 2', async () => {
    const user = userEvent.setup()
    await user.click(screen.getByTestId('parent-button-increment2'))
    expect(screen.getByTestId('parent-count2').innerText).toEqual('Count: 1')
})

test('useGlobalState -> read new count (child component)', async () => {
    expect(screen.getByTestId('child-count').innerText).toEqual('Count: 1')
})

test('useGlobalState -> read new count (child component 2)', async () => {
    expect(screen.getByTestId('child-count2').innerText).toEqual('Count: 1')
})

test('useGlobalState -> increment count (child component)', async () => {
    const user = userEvent.setup()
    await user.click(screen.getByTestId('child-button-increment'))
    expect(screen.getByTestId('child-count').innerText).toEqual('Count: 2')
})

test('useGlobalState -> increment count (child component 2)', async () => {
    const user = userEvent.setup()
    await user.click(screen.getByTestId('child-button-increment2'))
    expect(screen.getByTestId('child-count2').innerText).toEqual('Count: 2')
})

test('useGlobalState -> snapshot', async () => {
    expect(screen.getByTestId('parent-container')).toMatchSnapshot()
})