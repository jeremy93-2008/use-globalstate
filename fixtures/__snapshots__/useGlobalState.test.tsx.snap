// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGlobalState -> snapshot 1`] = `
<div
  data-testid="parent-container"
>
  <h1>
    Parent Component
  </h1>
  <p
    data-testid="parent-count"
  >
    Count: 
    2
  </p>
  <p
    data-testid="parent-count2"
  >
    Count: 
    2
  </p>
  <button
    data-testid="parent-button-increment"
  >
    Increment
  </button>
  <button
    data-testid="parent-button-increment2"
  >
    Increment
  </button>
  <div>
    <h1>
      Child Component
    </h1>
    <button
      data-testid="child-button-increment"
    >
      Increment
    </button>
    <p
      data-testid="child-count"
    >
      Count: 
      2
    </p>
    <button
      data-testid="child-button-increment2"
    >
      Increment
    </button>
    <p
      data-testid="child-count2"
    >
      Count: 
      2
    </p>
  </div>
</div>
`;
